warehouse/query-core/src/main/java/datawave/query/iterator/logic/TermFrequencyExcerptIterator.java

```diff
@@ -315,28 +386,225 @@ public void next() throws IOException {
+            } else {
+                // join everything together with spaces while skipping null offsets (after trimming both sides down to what we need)
+                return joiner.join(bothTrim(termsToOutput));
+            }
+        } else {
+            if (bef) { // if direction is "before", set everything after the last hit term to null
+                for (int k = beforeIndex + 1; k < terms.length; k++) {
+                    termsToOutput[k] = null;
+                }
+                if (trim) {
+                    float start = beforeIndex - (origHalfSize * 2);
+                    trimBeginning(termsToOutput, beforeIndex, start);
+                }
+                // join everything together with spaces while skipping null offsets
+                return joiner.join(termsToOutput);
+            } else if (aft) { // if direction is "after", set everything before the first hit term to null
+                for (int k = 0; k < afterIndex; k++) {
+                    termsToOutput[k] = null;
+                }
+                if (trim) {
+                    float start = afterIndex + (origHalfSize * 2);
+                    trimEnd(termsToOutput, afterIndex, start);
+                }
+                // join everything together with spaces while skipping null offsets
+                return joiner.join(termsToOutput);
+            } else {
+                if (!trim) {
+                    // join everything together with spaces while skipping null offsets
+                    return joiner.join(termsToOutput);
+                } else {
+                    // join everything together with spaces while skipping null offsets (after trimming both sides down to what we need)
+                    return joiner.join(bothTrim(termsToOutput));
+                }
+            }
+        }
+    }
 
-        return joiner.join(largestTerms);
+    /**
+     * Trim down both side of the excerpt to the size that we want
+     *
+     * @param termsToOutput
+     *            the terms to create a phrase from
+     * @return the trimmed array
+     */
+    private String[] bothTrim(String[] termsToOutput) {
+        int expandedMid = (endOffset - startOffset) / 2;
+        int start = (int) (expandedMid - origHalfSize);
+        trimBeginning(termsToOutput, expandedMid, start);
+        start = (int) (expandedMid + origHalfSize);
+        trimEnd(termsToOutput, expandedMid, start);
+        return termsToOutput;
     }
 
     /**
-     * Get the longest term from a list of terms;
+     * Trims off the front to get the correct size
+     *
+     * @param termsToOutput
+     *            the terms to create a phrase from
+     * @param beforeIndex
+     *            the index in the array to start at
+     * @param start
+     *            the index in the array where we want to start setting values to null at
+     */
+    private void trimBeginning(String[] termsToOutput, int beforeIndex, float start) {
```

Comment:
Should `start` be `int` rather than `float`?

Reply:
possibly. I don't remember why I made it a float in the first place. Just have to cast the computation for "start" to an int to get everything to compile still.